tests/unit/test_handlers.py

```diff
@@ -1944,3 +1917,80 @@ def test_document_response_params_without_expires(document_expires_mocks):
     mocks['section'].get_section.assert_not_called()
     mocks['param_section'].add_new_section.assert_not_called()
     mocks['doc_section'].write.assert_not_called()
+
+
+@pytest.fixture()
+def operation_model_for_200_error():
+    operation_model = mock.Mock()
+    operation_model.has_streaming_output = False
+    return operation_model
+
+
+@pytest.fixture()
+def response_dict_for_200_error():
+    return {
+        'status_code': 200,
+        'body': (
+            b"<Error>"
+            b"<Code>SlowDown</Code>"
+            b"<Message>Please reduce your request rate.</Message>"
+            b"</Error>"
+        ),
+    }
+
+
+def test_500_status_code_set_for_200_response(
+    operation_model_for_200_error, response_dict_for_200_error
+):
+    handlers._handle_200_error(
+        operation_model_for_200_error, response_dict_for_200_error
+    )
+    assert response_dict_for_200_error['status_code'] == 500
+
+
+def test_200_response_with_no_error_left_untouched(
+    operation_model_for_200_error, response_dict_for_200_error
+):
+    response_dict_for_200_error['body'] = b"<NotAnError/>"
+    handlers._handle_200_error(
+        operation_model_for_200_error, response_dict_for_200_error
+    )
+    # We don't touch the status code since there are no errors present.
+    assert response_dict_for_200_error['status_code'] == 200
+
+
+def test_200_response_with_streaming_output_left_untouched(
+    operation_model_for_200_error,
+    response_dict_for_200_error,
+):
+    operation_model_for_200_error.has_streaming_output = True
+    handlers._handle_200_error(
+        operation_model_for_200_error, response_dict_for_200_error
+    )
+    # We don't touch the status code on streaming operations.
+    assert response_dict_for_200_error['status_code'] == 200
+
+
+def test_200_response_with_no_body_left_untouched(
+    operation_model_for_200_error, response_dict_for_200_error
+):
+    response_dict_for_200_error['body'] = b""
+    handlers._handle_200_error(
+        operation_model_for_200_error, response_dict_for_200_error
+    )
+    assert response_dict_for_200_error['status_code'] == 200
+
+
+def test_http_status_code_updated_to_retry_200_response():
+    http_response = mock.Mock()
+    http_response.status_code = 200
+    parsed = {}
+    parsed.setdefault('ResponseMetadata', {})['HTTPStatusCode'] = 500
+    handlers._retry_200_error((http_response, parsed))
+    assert http_response.status_code == 500
+
+
+def test_500_response_can_be_none():
+    # A 500 response can raise an exception, which means the response
+    # object is None.  We need to handle this case.
+    handlers._retry_200_error(None)
```

Comment:
For future reference, we should always be asserting _something_ in our tests to ensure accidental no-op regressions don't sneak in. If we wanted to validate that `None` is a valid, we should be testing what the behavior of the function is when that's passed. In this case, we know it will return `None` if it's working as expected.

Read the above comment about testing private functions though before making any changes.